frontend/packages/dev-console/src/components/pipelines/pipeline-builder/task-sidebar/TaskSidebarParam.tsx

```diff
@@ -13,31 +14,36 @@ const TaskSidebarParam: React.FC<TaskSidebarParamProps> = (props) => {
   const { hasParamError, onChange, resourceParam, taskParam } = props;
   const [dirty, setDirty] = React.useState(false);
 
-  const currentValue = taskParam?.value?.toString() || '';
+  const currentValue = taskParam?.value;
   const emptyIsInvalid = !resourceParam.default;
 
-  const isValid = !(dirty && hasParamError && emptyIsInvalid && currentValue === '');
+  const isValid = !(dirty && hasParamError && emptyIsInvalid && currentValue != null);
+
+  const paramRenderProps: ParameterProps = {
+    currentValue,
+    defaultValue: resourceParam.default,
+    isValid,
+    name: resourceParam.name,
+    onChange,
+    setDirty,
+  };
 
   return (
     <FormGroup
       fieldId={resourceParam.name}
       label={resourceParam.name}
-      helperText={resourceParam.description}
+      helperText={resourceParam.type === 'string' ? resourceParam.description : null}
```

Comment:
is it by design that there aint helpertext for array? I dont see it at an individual param level as well

Reply:
Yup, definitely is... however due to the way the errors are handled (outside of yup) it's not possible to use Formik for this solution as it would subvert all the error handling right now. That's why there is a refactor needed as part of ODC-3165.

There may need additional design changes come the refactor, but until then this is the best we have - a hack to merge the underlying components together to create this one.

Reply:
/LGTM